riscv/processor.h

```diff
@@ -381,6 +385,7 @@ class processor_t : public abstract_device_t
   bool histogram_enabled;
   bool log_commits_enabled;
   FILE *log_file;
+  trace_encoder_n *trace_encoder;
```

Comment:
I think the member here does not need to be a pointer.

Reply:
Yes, the current code has a memory leak, and the easiest fix is to make this field a `trace_encoder_n` rather than a `trace_encoder_n *`.

Reply:
this is closely matching the implementation of some other processor fields like MMU. Also, using the cpp class construction `new trace_encoder_n();` returns a pointer, so I think this makes sense.

Reply:
I don't think the MMU pattern is necessarily the right one to follow here. We aren't consistent at all - note that `vectorUnit_t`, `entropy_source`, and `triggers::module_t` are not pointers.

Generally, you should use pointers if the member object is "given" to the class, rather than "created" by the class. We are obviously breaking this guidline for the mmu.

Reply:
Actually, the trace_encoder_ctrl device (https://github.com/ucb-bar/spike-devices/blob/46ba028b63ded301f8e2ad3bf1b9c51d5f46ab4e/src/trace_encoder_ctrl.cc#L44) needs a reference to the processor's trace encoder. Would this change the design decision here?